From 9112b916da046c06f614f67c4b5b1dcccb8792d3 Mon Sep 17 00:00:00 2001
From: Gleb Natapov <gleb@redhat.com>
Date: Wed, 12 Jan 2011 12:19:48 -0200
Subject: [PATCH 10/29] Populate drive_g->desc prior to calling add_bcv_internal().

RH-Author: Gleb Natapov <gleb@redhat.com>
Message-id: <1294834804-16829-10-git-send-email-gleb@redhat.com>
Patchwork-id: 16192
O-Subject: [PATCH SEABIOS RHEL6.1 09/25] Populate drive_g->desc prior to calling
	add_bcv_internal().
Bugzilla: 643688
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>

From: Kevin O'Connor <kevin@koconnor.net>

Make sure the description is populated before registering a drive.

Upstream commit: ce24be5

Signed-off-by: Gleb Natapov <gleb@redhat.com>
---
 src/usb-msc.c    |    5 ++---
 src/virtio-blk.c |    8 +++-----
 2 files changed, 5 insertions(+), 8 deletions(-)

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 src/usb-msc.c    |    5 ++---
 src/virtio-blk.c |    8 +++-----
 2 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/src/usb-msc.c b/src/usb-msc.c
index 968cae3..9b146cf 100644
--- a/src/usb-msc.c
+++ b/src/usb-msc.c
@@ -232,6 +232,8 @@ usb_msc_init(struct usb_pipe *pipe
             , strtcpy(rev, data.rev, sizeof(rev))
             , pdt, removable);
     udrive_g->drive.removable = removable;
+    snprintf(desc, MAXDESCSIZE, "USB Drive %s %s %s", vendor, product, rev);
+    udrive_g->drive.desc = desc;
 
     if (pdt == USB_MSC_TYPE_CDROM)
         ret = setup_drive_cdrom(&dop);
@@ -240,9 +242,6 @@ usb_msc_init(struct usb_pipe *pipe
     if (ret)
         goto fail;
 
-    snprintf(desc, MAXDESCSIZE, "USB Drive %s %s %s", vendor, product, rev);
-    udrive_g->drive.desc = desc;
-
     return 0;
 fail:
     dprintf(1, "Unable to configure USB MSC device.\n");
diff --git a/src/virtio-blk.c b/src/virtio-blk.c
index 7a25826..9c9ed83 100644
--- a/src/virtio-blk.c
+++ b/src/virtio-blk.c
@@ -151,15 +151,13 @@ init_virtio_blk(u16 bdf)
     vdrive_g->drive.pchs.cylinders = cfg.cylinders;
     vdrive_g->drive.pchs.heads = cfg.heads;
     vdrive_g->drive.pchs.spt = cfg.sectors;
-
-    setup_translation(&vdrive_g->drive);
-    add_bcv_internal(&vdrive_g->drive);
-
     snprintf(desc, MAXDESCSIZE, "Virtio disk PCI:%x:%x",
              pci_bdf_to_bus(bdf), pci_bdf_to_dev(bdf));
-
     vdrive_g->drive.desc = desc;
 
+    setup_translation(&vdrive_g->drive);
+    add_bcv_internal(&vdrive_g->drive);
+
     vp_set_status(ioaddr, VIRTIO_CONFIG_S_ACKNOWLEDGE |
                   VIRTIO_CONFIG_S_DRIVER | VIRTIO_CONFIG_S_DRIVER_OK);
     return;
-- 
1.7.4.rc1.16.gd2f15e

